### PR TITLE
Documentation for Autest in Sphinx

### PR DESCRIPTION
**Pull request by macisasandwich**
Moved the old README under tests/ to docs. Part 1 of few commits in a row to bring the testing docs up-to-date.

### CODE REVIEW

File: doc/developer-guide/testing/blackbox-testing.en.rst [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+A number of file objects are also defined to help test TrafficServer. Files that are currently defined are:
+
+- squid.log 
+- error.log
+- diags.log
+- records.config
+- cache.config
+- congestion.config
+- hosting.config
+- ip_allow.config
```
Comment (randall) [inline doc/developer-guide/testing/blackbox-testing.en.rst]: this is now ip_allow.yaml

Comment (macisasandwich) [reply doc/developer-guide/testing/blackbox-testing.en.rst]: fixed

**Comment by macisasandwich** [comment]
can I get a re-run on autest? Doc change doesn't affect autest failures ._.

**Comment by maskit** [comment]
[approve ci autest]

**Pull request merged**